pkg/infrastructure/azure/azure.go

```diff
@@ -26,3 +47,312 @@ func InitializeProvider() clusterapi.Provider {
+
+	// XXXX remove these when done testing
+	storageURL := fmt.Sprintf("https://%s.blob.core.windows.net", storageAccountName)
+	blobURL := fmt.Sprintf("%s/%s/%s", storageURL, containerName, blobName)
+
+	// Create resource group
+	resourcesClientFactory, err := armresources.NewClientFactory(
+		subscriptionID,
+		tokenCredential,
+		&arm.ClientOptions{
+			ClientOptions: policy.ClientOptions{
+				Cloud: cloudConfiguration,
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get azure resource groups factory: %w", err)
+	}
+	resourceGroupsClient := resourcesClientFactory.NewResourceGroupsClient()
+	resourceGroup, err := resourceGroupsClient.CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		armresources.ResourceGroup{
+			Location:  ptr.To(platform.Region),
+			ManagedBy: nil,
+			Tags:      tags,
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("error creating resource group %s: %w", resourceGroupName, err)
+	}
+	logrus.Debugf("ResourceGroup.ID=%s", *resourceGroup.ID)
+
+	// Create user assigned identity
+	userAssignedIdentityName := fmt.Sprintf("%s-identity", in.InfraID)
+	armmsiClientFactory, err := armmsi.NewClientFactory(subscriptionID, tokenCredential, nil)
+	if err != nil {
+		log.Fatalf("failed to create armmsi client: %v", err)
+	}
+	userAssignedIdentity, err := armmsiClientFactory.NewUserAssignedIdentitiesClient().CreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		userAssignedIdentityName,
+		armmsi.Identity{
+			Location: ptr.To(platform.Region),
+			Tags:     tags,
+		},
+		nil,
+	)
+	if err != nil {
+		log.Fatalf("failed to create user assigned identity %s: %v", userAssignedIdentityName, err)
+	}
+	principalID := *userAssignedIdentity.Properties.PrincipalID
+
+	logrus.Debugf("UserAssignedIdentity.ID=%s", *userAssignedIdentity.ID)
+	logrus.Debugf("PrinciapalID=%s", principalID)
+
+	// Create role assignments
+	resourceGroupScope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, resourceGroupName)
+
+	clientFactory, err := armauthorization.NewClientFactory(
+		subscriptionID,
+		tokenCredential,
+		&arm.ClientOptions{
+			ClientOptions: policy.ClientOptions{
+				Cloud: cloudConfiguration,
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to get authorization factory: %w", err)
+	}
+
+	roleDefinitionsClient := clientFactory.NewRoleDefinitionsClient()
+	roleDefinitionsPager := roleDefinitionsClient.NewListPager(resourceGroupScope, nil)
+
+	var roleDefinitionBlobOwner *armauthorization.RoleDefinition
+	for roleDefinitionsPager.More() {
+		rolePage, err := roleDefinitionsPager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get role definition page: %w", err)
+		}
+		for _, roleDefinition := range rolePage.Value {
+			if *roleDefinition.Properties.RoleName == "Storage Blob Data Owner" {
+				roleDefinitionBlobOwner = roleDefinition
+				break
+			}
+		}
+	}
+	if roleDefinitionBlobOwner == nil {
+		return fmt.Errorf("failed to find storage blob data owner role")
+	}
```

Comment:
This seems unused. Not sure if we need this?

Reply:
Fixed.